### PR TITLE
Make getRuntimeInfo() method async

### PR DESCRIPTION
**Pull request by vinokurig**
Is needed for https://github.com/eclipse/che/issues/11954

### CODE REVIEW

File: packages/task/src/node/process/process-task.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class ProcessTask extends Task {
         });
     }
 
-    getRuntimeInfo(): ProcessTaskInfo {
+    async getRuntimeInfo(): Promise<ProcessTaskInfo> {
```
Comment (elaihau) [inline packages/task/src/node/process/process-task.ts]: this function does not need to be `async` i guess ?

Comment (elaihau) [reply packages/task/src/node/process/process-task.ts]: ahh, i see this line: https://github.com/theia-ide/theia/blob/master/packages/task/src/node/task-runner.ts#L32

just thinking: maybe we should change the return type of the function in the interface from `Promise<...>` to `MaybePromise<...>` ?

Comment (vinokurig) [reply packages/task/src/node/process/process-task.ts]: fixed

Comment (vinokurig) [reply packages/task/src/node/process/process-task.ts]: @elaihau could you please review my fix

Comment (elaihau) [reply packages/task/src/node/process/process-task.ts]: @vinokurig sorry i still don't understand why updating `getRuntimeInfo()` is related to https://github.com/eclipse/che-theia-task-plugin/blob/master/che-theia-task-extension/src/node/che-task-runner.ts#L21

Could you please kindly explain ?

Comment (vinokurig) [reply packages/task/src/node/process/process-task.ts]: I need it for my PR: https://github.com/eclipse/che-theia/blob/09515ef56850be2186afa23ba1188945682f5a72/extensions/eclipse-che-theia-plugin-ext/src/node/task-service.ts#L77

Comment (elaihau) [reply packages/task/src/node/process/process-task.ts]: Got it. I am not an expert of `Promise` - if I were you, I would update
[this line](https://github.com/eclipse/che-theia/blob/09515ef56850be2186afa23ba1188945682f5a72/extensions/eclipse-che-theia-plugin-ext/src/node/task-service.ts#L77) in your other PR into:
```ts
async getRuntimeInfo(): Promise<TaskInfo> {
```
and revert the change made to [this line](https://github.com/theia-ide/theia/pull/3909/files#diff-bccc335bccb5d8638a4987cf68ee36b3R73) in this PR.

You could check if what I suggested works.

Comment (vinokurig) [reply packages/task/src/node/process/process-task.ts]: fixed
@elaihau Thank you for your advise, it works.

**Pull request merged**